amundsen_application/static/js/components/common/ResourceList/PaginatedResourceList/index.spec.tsx

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import Pagination from 'react-js-pagination';
+
+import { shallow } from 'enzyme';
+
+import { ResourceType } from 'interfaces';
+import ResourceListItem from 'components/common/ResourceListItem/index';
+
+import PaginatedResourceList, { PaginatedResourceListProps } from './';
+import * as CONSTANTS from '../constants';
+
+describe('PaginatedResourceList', () => {
+  const setStateSpy = jest.spyOn(PaginatedResourceList.prototype, 'setState');
+  const setup = (propOverrides?: Partial<PaginatedResourceListProps>) => {
+    const props: PaginatedResourceListProps = {
+      allItems: [
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+        { type: ResourceType.table },
+      ],
+      itemsPerPage: 4,
+      source: 'testSource',
+      ...propOverrides
+    };
+    const wrapper = shallow<PaginatedResourceList>(<PaginatedResourceList {...props} />);
+    return { props, wrapper };
+  };
+
+  describe('componentDidUpdate', () => {
+    it('updates activePage state if the activePage is out of bounds by multiple pages', () => {
+      const wrapper = setup({ itemsPerPage: 2 }).wrapper;
+      wrapper.setState({ activePage: 2 });
+      const componentDidUpdateSpy = jest.spyOn(wrapper.instance(), 'componentDidUpdate');
+      wrapper.setProps({
+        allItems: [
+          { type: ResourceType.table },
+          { type: ResourceType.table },
+        ]
+      });
+      expect(componentDidUpdateSpy).toHaveBeenCalled();
```

Comment:
I am not sure of the utility of this expectation. Wouldn't this be called anyways?

Reply:
That's fair, the valuable assertion to me here is `expect(wrapper.state().activePage).toEqual(0);`